orm/src/test/java/org/hibernate/search/test/query/SortTest.java

```diff
@@ -272,6 +278,67 @@ public void testResultOrderedByEmbeddedAuthorNameAscending() throws Exception {
 		tx.commit();
 	}
 
+	@Test
+	public void testCombinedQueryOnIndexWithSortFieldAndIndexToBeUninverted() throws Exception {
+		Transaction tx = fullTextSession.beginTransaction();
+
+		Query query = queryParser.parse( "name:Bill" );
+		FullTextQuery hibQuery = fullTextSession.createFullTextQuery( query, Plumber.class, BrickLayer.class );
+		Sort sort = new Sort( new SortField( "sortName", SortField.Type.STRING ) ); //ASC
+		hibQuery.setSort( sort );
+
+		@SuppressWarnings("unchecked")
+		List<Book> result = hibQuery.list();
+		assertNotNull( result );
+		assertThat( result ).onProperty( "name" )
+			.describedAs( "Expecting results from index with sort field and uninverted index in the correct sort order" )
+			.containsExactly( "Bill the brick layer", "Bill the plumber" );
+
+		tx.commit();
+	}
+
+	@Test
+	public void testQueryOnIndexSharedByEntityWithRequiredSortFieldAndEntityWithoutRaisesException() throws Exception {
+		Transaction tx = fullTextSession.beginTransaction();
+
+		Query query = queryParser.parse( "name:Bill" );
+		FullTextQuery hibQuery = fullTextSession.createFullTextQuery( query, Thatcher.class, BrickLayer.class );
```

Comment:
What's the difference between `Thatcher` and `Plumber`?
As far as I understand they both have a sortName field?

I don't understand why the previous test passes while this one fails.